subsystem/core/src/main/java/org/wildfly/extension/camel/service/CamelEndpointDeployerService.java

```diff
@@ -0,0 +1,497 @@
+        }
+
+        public String relativize(String path) {
+            final StringTokenizer patternTokenizer = new StringTokenizer(source, "/", true);
+            final StringTokenizer pathTokenizer = new StringTokenizer(path, "/", true);
+            /* skip the common segments */
+            while (pathTokenizer.hasMoreTokens()) {
+                if (!patternTokenizer.hasMoreTokens()) {
+                    return null;
+                }
+                final String patternToken = patternTokenizer.nextToken();
+                final String pathToken = pathTokenizer.nextToken();
+                final boolean isSlash = patternToken.equals("/");
+                if (isSlash != pathToken.equals("/")) {
+                    return null;
+                } else if (!isSlash) {
+                    if ("*".equals(pathToken)) {
+                        continue;
+                    } else {
+                        final StringTokenizer segmentTokenizer = new StringTokenizer(patternToken, "*", true);
+                        final StringBuilder segmentPatternBuilder = new StringBuilder(patternToken.length());
+                        while (segmentTokenizer.hasMoreTokens()) {
+                            final String segmentToken = segmentTokenizer.nextToken();
+                            if ("*".equals(segmentToken)) {
+                                segmentPatternBuilder.append(".*");
+                            } else {
+                                segmentPatternBuilder.append(Pattern.quote(segmentToken));
+                            }
+                        }
+                        final Pattern pat = Pattern.compile(segmentPatternBuilder.toString());
+                        if (!pat.matcher(pathToken).matches()) {
+                            return null;
+                        }
+                    }
+                }
+            }
+            final StringBuilder result = new StringBuilder();
+            while (patternTokenizer.hasMoreTokens()) {
+                result.append(patternTokenizer.nextToken());
+            }
+            if (result.length() == 0) {
+                return "";
+            } else if (result.charAt(0) != '/') {
+                return "/" + result.toString();
+            } else {
+                return result.toString();
+            }
+        }
+
+    }
+
+    private static final String CATCH_ALL_ENDPOINT_URI_PREFIX = "///*";
+
+    private static final String CATCH_ALL_PREFIX = "/*";
+
+    /** The name for the {@link CamelEndpointDeployerService} */
+    private static final String SERVICE_NAME = "EndpointDeployer";
+
+    /**
+     * This method can simplified substantially, once https://github.com/undertow-io/undertow/pull/642 reaches us.
+     * Currently, the method is just an adjusted copy of {@link DeploymentInfo#clone()}.
+     *
+     * @param securityConstraints
+     * @param servletInfo
+     * @param deploymentName
+     * @param contextPath
```

Comment:
The `@param` documentation does not seem to be correct for the method signature.

Reply:
@jamesnetherton has eagle's eyes!